File: fixtures/scale/ruby/large/lib/check_stridecode.rb
require_relative 'stridecode_encode'
require_relative 'stridecode_decode'
require_relative 'corpus_codec'

def check_stridecode
  corpus_codec.each do |text|
    packed = stridecode_encode(text)
    return false if stridecode_decode(packed) != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_stridecode.py
from cipherlab.ciphers.stridecode import stridecode_encode, stridecode_decode
from cipherlab.specs.stridecode_spec import stridecode_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_stridecode():
    spec = stridecode_spec()
    for text in corpus_codec():
        packed = stridecode_encode(text)
        if stridecode_decode(packed) != text:
            return False
    return spec["category"] == "codec"
